tiledb/sm/subarray/subarray.cc

```diff
@@ -555,6 +837,95 @@ Status Subarray::get_est_result_size(
   return Status::Ok();
 }
 
+Status Subarray::get_est_result_size_querytype_audited(
+    const char* name, uint64_t* size, StorageManager* storage_manager) {
+  QueryType type;
+  // Note: various items below expect array open, get_query_type() providing
+  // that audit.
+  RETURN_NOT_OK(array_->get_query_type(&type));
+
+  if (type == QueryType::WRITE)
+    return LOG_STATUS(Status::QueryError(
+        "Cannot get estimated result size; Operation currently "
+        "unsupported for write queries"));
+
+  if (name == nullptr)
+    return LOG_STATUS(Status::QueryError(
+        "Cannot get estimated result size; Name cannot be null"));
+
+  if (name == constants::coords &&
+      !array_->array_schema()->domain()->all_dims_same_type())
+    return LOG_STATUS(Status::QueryError(
+        "Cannot get estimated result size; Not applicable to zipped "
+        "coordinates in arrays with heterogeneous domain"));
+
+  if (name == constants::coords &&
+      !array_->array_schema()->domain()->all_dims_fixed())
+    return LOG_STATUS(Status::QueryError(
+        "Cannot get estimated result size; Not applicable to zipped "
+        "coordinates in arrays with domains with variable-sized dimensions"));
+
+  if (array_->array_schema()->is_nullable(name))
+    return LOG_STATUS(Status::WriterError(
+        std::string(
+            "Cannot get estimated result size; Input attribute/dimension '") +
+        name + "' is nullable"));
+
+  if (array_->is_remote()) {
+#if 01
```

Comment:
Let's figure this out before we merge and remove the #if

Reply:
subarray.cc and comment regarding added to ch7841